packages/dds/tree/src/simple-tree/prepareForInsertion.ts

```diff
@@ -40,14 +43,16 @@ import { debugAssert, oob } from "@fluidframework/core-utils/internal";
  */
 export function prepareForInsertion<TIn extends InsertableContent | undefined>(
 	data: TIn,
-	schema: ImplicitFieldSchema,
+	schema: ImplicitAnnotatedFieldSchema,
 	destinationContext: FlexTreeContext,
+	allowNonEnabledTypes = false,
```

Comment:
I don't think it makes sense for this to be a parameter here (or for any of the public function sin this file).

The job of this file is to be a uniform place to enforce consistent handling of insertions. There is no reason we would want different places calling this function to be have differently. It should make the choice internally based on some policy.

I'm not clear what that policy is currently supposed to be (are they enabled in unhydrated contexts? When actually hydrating do we unconditionally disable them?). This code has enough information to make those choices and implement those polices so it should be done here, in my opinion.